src/lib.rs

```diff
@@ -1264,6 +1277,61 @@ where
     }
 }
 
+/// A streaming iterator that flattens nested streaming iterators.
+#[derive(Debug)]
+pub struct Flatten<I, J> {
+    iter: I,
+    _inner: PhantomData<J>,
+}
```

Comment:
I don't think we need the `J` parameter at all, and then you don't need to phantom anything.

(The standard library also only has `Flatten<I>`.)